builder/assignmentEnv.go

```diff
@@ -0,0 +1,315 @@
+package builder
+
+import (
+	"assignment-exec/image-builder/configurations"
+	"assignment-exec/image-builder/constants"
+	"assignment-exec/image-builder/environment"
+	"bytes"
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/client"
+	"github.com/pkg/errors"
+	"io"
+	"log"
+	"os"
+	"strings"
+)
+
+type baseEnv interface {
+	verifyAndWrite() error
+	build() error
+	publish() error
+	undoWrite()
+	undoBuild() error
+	undoPublish() error
+}
+
+type assignmentEnv struct {
```

Comment:
Might not be the best idea to use abbreviations for type names.
Abbreviations and acronyms are okay for variable names.

Reply:
okay. changed it to `assignmentEnvironment`